### ISSUE TITLE
[java] False Positive: PMD Incorrectly report violation for UnnecessaryImport

### ISSUE THREAD

**Issue by soyodream** [pmd/pmd#4785]
**Affects PMD Version:7.0.0-rc4** 

**Rule:UnnecessaryImport**

<https://docs.pmd-code.org/latest/pmd_rules_java_codestyle.html#unnecessaryimport>

**Description:**
According to the rule description, if a class is imported, and subsequently, used in the code, PMD should not report it as a violation. However, the provided sample code, adhering to the rule's specifications, is erroneously marked as a violation. This inconsistency suggests a false positive in the PMD analysis.
**Code Sample demonstrating the issue:**

```java
import org.junit.Test;
import java.util.Arrays;
import static org.junit.Assert.assertTrue;
public class Foo{
   @Test
   public void testfoo(Object a){
       assertTrue(a instanceof Arrays);
   }
}

```

**Expected outcome:**

PMD reports a violation at line 2 and line 3, but that's wrong. That's a false positive.

**Running PMD through:** *[CLI]*

**Comment by julius-d** [comment]
Hi!
While updating a project to PMD 7, I found a similar false positive:
```java


import static java.util.Comparator.comparing;

import java.util.List;
import java.util.Locale;
import java.util.Optional;

public class Example {

  public Optional<String> getBar(List<String> foo) {
    return foo.stream()
        .sorted(comparing((String it) -> it.toLowerCase(Locale.ROOT).trim()))
        .findAny();
  }
}

```
PMD output:
```
[INFO] PMD Failure: Example:3 Rule:UnnecessaryImport Priority:4 
Unused static import 'java.util.Comparator.comparing'.
```
I am using  `PMD version: 7.0.0` via the `maven-pmd-plugin:3.21.2`

**Comment by oowekyala** [comment]
> Hi!
> While updating a project to PMD 7, I found a similar false positive:


Note: #4920 will fix this issue. But the underlying issue is that overload resolution failed for the call to `comparing`. The code is well formed, but the explicitly typed lambda is not yet supported by our type resolution engine:

https://github.com/pmd/pmd/blob/280b1a1ee34dc0ae35e0ac7d578233b637942ef4/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/types/internal/infer/ExprCheckHelper.java#L585-L588

In the meantime you can remove the explicit argument type in `(String it) -> ...` to fix the warning.
